packages/twenty-server/src/database/commands/upgrade-version/0-23/0-23-update-file-folder-structure.command.ts

```diff
@@ -0,0 +1,231 @@
+import { Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import chalk from 'chalk';
+import { Command, CommandRunner, Option } from 'nest-commander';
+import { Like, Repository } from 'typeorm';
+
+import { FileFolder } from 'src/engine/core-modules/file/interfaces/file-folder.interface';
+
+import { TypeORMService } from 'src/database/typeorm/typeorm.service';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { FileStorageService } from 'src/engine/integrations/file-storage/file-storage.service';
+import { DataSourceService } from 'src/engine/metadata-modules/data-source/data-source.service';
+
+interface UpdateFileFolderStructureCommandOptions {
+  workspaceId?: string;
+}
+
+@Command({
+  name: 'upgrade-0-23:update-file-folder-structure',
+  description: 'Update file folder structure (prefixed per workspace)',
+})
+export class UpdateFileFolderStructureCommand extends CommandRunner {
+  private readonly logger = new Logger(UpdateFileFolderStructureCommand.name);
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    private readonly workspaceRepository: Repository<Workspace>,
+    private readonly typeORMService: TypeORMService,
+    private readonly dataSourceService: DataSourceService,
+    private readonly fileStorageService: FileStorageService,
+  ) {
+    super();
+  }
+
+  @Option({
+    flags: '-w, --workspace-id [workspace_id]',
+    description: 'workspace id. Command runs on all workspaces if not provided',
+    required: false,
+  })
+  parseWorkspaceId(value: string): string {
+    return value;
+  }
+
+  async run(
```

Comment:
I know that this method is really verbose, but it felt like such one off script is better to be understandable rather than introducing unnecessary abstraction. Let me know what you think.